railties/lib/rails/commands/console/irb_console.rb

```diff
@@ -87,7 +87,8 @@ def start
 
         env = colorized_env
         prompt_prefix = "%N(#{env})"
-        IRB.conf[:IRB_NAME] = @app.name
+        # Respect user's configured irb name.
+        IRB.conf[:IRB_NAME] = @app.name if IRB.conf[:IRB_NAME] == "irb"
```

Comment:
This seems likely to break if irb ever changes the default name.

I'm not sure we need to respect their setting here either, as the goal seems to be to put the Application name front and center to avoid confusion.

While this change could re-introduce that ambiguity.

Reply:
Seems unlikely that they'd change the default name.

That being said, this is as prone to issues as the PROMPT_MODE config option.